Dockerfile

```diff
@@ -35,9 +36,17 @@ RUN --mount=target=. \
     -o /go/bin/goshimmer
 
 # Enable building the image without downloading the snapshot.
+# It's possible to download custom snapshot from external storage service - necessary for feature network deployment.
 # If built with dummy snapshot then a snapshot needs to be mounted into the resulting image.
-RUN if [ $DOWNLOAD_SNAPSHOT -gt 0 ]; then \
+RUN if [ $DOWNLOAD_SNAPSHOT -gt 0 ] && [ "$CUSTOM_SNAPSHOT" == "false" ]; then \
     wget -O /tmp/snapshot.bin https://dbfiles-goshimmer.s3.eu-central-1.amazonaws.com/snapshots/nectar/snapshot-latest.bin ;  \
+    elif [ $DOWNLOAD_SNAPSHOT -gt 0 ] && [ "$CUSTOM_SNAPSHOT" != "false" ]; then \
```

Comment:
Have you tested that all those conditions work as expected in all scenarios?
We with @jonastheis were having problems with the `[ $DOWNLOAD_SNAPSHOT -gt 0 ]` condition before.

Reply:
No. Will do that!

Reply:
Also make sure that, if necessary, the cache is invalidated. It could be that this layer/step is cached even though the parameters changed which could lead to undesired effects: the wrong snapshot being included.